modules/auxiliary/scanner/smb/smb_enumshares.rb

```diff
@@ -76,39 +81,37 @@ def enum_tree(tree, share, subdir = '')
     write = tree.permissions.write_ea == 1
     skip = false
 
-    if ENUMERABLE_SHARE_TYPES.include? share[:type]
+    if ENUMERABLE_SHARE_TYPES.include?(share[:type])
       msg = share[:type]
-    elsif SKIPPABLE_SHARE_TYPES.include? share[:type]
+    elsif SKIPPABLE_SHARE_TYPES.include?(share[:type])
       msg = share[:type]
       skip = true
     else
       msg = "Unhandled Device Type (#{share[:type]})"
       skip = true
     end
 
-    print_status "Skipping share #{share[:name].strip} as it is of type #{share[:type]}" if skip
+    print_status("Skipping share #{share[:name].strip} as it is of type #{share[:type]}") if skip
     return read, write, msg, nil if skip
 
     # Create list after possibly skipping a share we wouldn't be able to access.
     begin
       list = tree.list(directory: subdir)
     rescue RubySMB::Error::UnexpectedStatusCode => e
-      print_error e.to_s
-      return read, write, msg, nil
+      vprint_error("Error when trying to list tree contents in #{share[:name]}\\#{subdir} - #{e.status_code.name}")
```

Comment:
So this is a nicer version of the error but now you don't have any return code here. Whilst you have a check later on, its not really complete.

Reply:
After some testing on Windows XP where I got access to the hidden `C$` share, I can't return here as it would stop the module from iterating over the whole share.
The module can receive the following (with `set verbose true`):
```
[-] 192.168.146.142:445   - Error when trying to list tree contents in C$\Program Files - STATUS_INVALID_HANDLE
[-] 192.168.146.142:445   - Error when trying to list tree contents in C$\WINDOWS\inf - STATUS_INVALID_HANDLE
[-] 192.168.146.142:445   - Error when trying to list tree contents in C$\WINDOWS\Media - STATUS_INVALID_HANDLE
```
However we are still able to enumerate files in other directories.
I'll add a better check after this function returns but I think there should be no return statement here. Correct me if I'm wrong 👍 
Unless I return unless the error we receive is `STATUS_INVALID_HANDLE`? 🤔

Reply:
You're right, apologies 👍 
I was under the wrong impression that having a `return read, write, msg, []` here would result in us returning from iterating over the rest of the dirs/files in that share but that is not the case.
I'll add this return back in the next commit. Thanks!

Reply:
Should this not be `nil` instead of `[]` to keep in line with the `return read, write, msg, nil if skip` line?

Reply:
Resolved in ce3215e35b83ac9cb61b08d82b498d44354589ec